### PR TITLE
Fix / Filter issue on navigation without clearing

### PR DESCRIPTION
**Pull request by thekevinbrown**
Issue: https://jam.dev/c/aeab7cfe-e767-4a88-9dd9-f09ebde4f317

* Switched to controlled inputs in all cases.
* Added tests for filters.
* Moved away from `initialValue` for components to `value`.

## Summary by CodeRabbit

- **New Features**
	- Enhanced the ComboBox component to support both single and multiple selection modes.
	- Introduced a new `useDataTransform` hook for managing data transformations.
	- Added support for media file upload and delete operations in MediaField.

- **Bug Fixes**
	- Corrected the handling of values in DetailPanel to strip off `__typename`.
	- Fixed z-index issue affecting layout positioning in the ComboBox component.

- **Tests**
	- Added test cases for filters to ensure proper functionality when clearing or using them.

### CODE REVIEW

File: src/packages/admin-ui-components/src/detail-panel/component.tsx [hunk 1]
```diff
@@ -53,9 +54,9 @@ const getField = ({ field, autoFocus }: { field: EntityField; autoFocus: boolean
 	if (field.relationshipType) {
 		// If the field is readonly and a relationship, show a link to the entity/entities
 		if (isReadonly) {
-			return <LinkField name={field.name} entity={field} />;
+			return <LinkField name={field.name} field={field} />;
 		}
-		return <RelationshipField name={field.name} entity={field} autoFocus={autoFocus} />;
+		return <RelationshipField name={field.name} field={field} autoFocus={autoFocus} />;
```
[the remark this answers is not in the document]
Comment (thekevinbrown) [reply src/packages/admin-ui-components/src/detail-panel/component.tsx:59]: LinkField can't be focused though. That prop doesn't exist.

File: src/packages/admin-ui-components/src/filters/relationship-filter.tsx [hunk 2]
```diff
@@ -37,6 +33,22 @@ export const RelationshipFilter = ({
...
 
+	const currentFilterValue =
+		(filter?.[fieldName] as Record<string, string[]> | undefined)?.[
+			`${relatedEntity.primaryKeyField}_in`
+		] ?? [];
+
+	// This reads the data for the related entity directly from the Apollo cache without going back to
+	// the server. The reason we always get the first one is we only display the name in the filter if there's
+	// one selected item. It will be in the cache because the grid will have fetched it.
+	const { data: displayData } = useFragment({
```
Comment (steve-keep) [inline src/packages/admin-ui-components/src/filters/relationship-filter.tsx:44]: `useFragment` === 🤯

**Review by steve-keep — APPROVED** [review]
This looks great!

File: src/packages/admin-ui-components/src/detail-panel/use-data-transform.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { useEffect } from 'react';
+import { EntityField } from '../utils';
+
+export interface DataTransform<T> {
+	field: EntityField;
+	transform: (value: T) => T | Promise<T>;
+}
+
+export const dataTransforms: DataTransform<unknown>[] = [];
```
[the remark this answers is not in the document]
Comment (thekevinbrown) [reply src/packages/admin-ui-components/src/detail-panel/use-data-transform.tsx:9]: Yeah good shout.

**Pull request merged**